website/blog/2023-05-01-package-exports-support.md

```diff
@@ -0,0 +1,229 @@
+:::
+
+## For app developers
+
+Package Exports can be enabled today, experimentally.
+
+- Imports against packages that depend on Package Exports features (such as [**Firebase**](https://www.npmjs.com/package/firebase) and [**Storybook**](https://www.npmjs.com/search?q=%40storybook)) should now work as designed.
+- React Native for Web projects using Metro will now be able to use the `"browser"` conditional export, removing the need for workarounds.
+
+Enabling Package Exports brings a few [edge-case breaking changes](#breaking-changes) that may affect specific projects, and which you can [test today](#validating-changes-in-your-project).
+
+**In a future React Native release, Package Exports will be enabled by default**. In a chicken-and-egg situation, React Native apps were previously a holdout for some packages to migrate to `"exports"` — or used our `"react-native"` root field escape hatch. Supporting these features in Metro will allow the ecosystem to move forward.
+
+### Enabling Package Exports (experimental)
+
+Package Exports can be enabled in your app's [**metro.config.js**](https://github.com/facebook/react-native/blob/0.72-stable/packages/react-native/template/metro.config.js) file via the [`resolver.unstable_enablePackageExports`](https://facebook.github.io/metro/docs/configuration/#unstable_enablepackageexports-experimental) option.
+
+```js
+const config = {
+  // ...
+  resolver: {
+    unstable_enablePackageExports: true,
+  },
+};
+```
+
+Metro exposes two further resolver options which configure how conditional exports behave:
+
+- [`unstable_conditionNames`](https://facebook.github.io/metro/docs/configuration/#unstable_conditionnames-experimental) — The set of [condition names](https://nodejs.org/docs/latest-v19.x/api/packages.html#community-conditions-definitions) to assert when resolving conditional exports. By default, we match `['require', 'import', 'react-native']`.
+- [`unstable_conditionsByPlatform`](https://facebook.github.io/metro/docs/configuration/#unstable_conditionsbyplatform-experimental) — The additional condition names to assert when resolving for a given platform target. By default, this matches `'browser'` when the platform is `'web'`.
+
+:::tip
+**Remember to use the React Native [Jest preset](https://github.com/facebook/react-native/blob/main/template/jest.config.js#L2)!** Jest includes support for Package Exports by default. In tests, you can override which `customExportConditions` are resolved using the [`testEnvironmentOptions`](https://jestjs.io/docs/configuration#testenvironmentoptions-object) option.
+:::
+
+#### Validating changes in your project
+
+For existing projects, we recommend that early adopters follow these steps to see if resolution changes occur after enabling `unstable_enablePackageExports`. This is a one-time process. It's likely that there will be no changes at all, but we'd like developers to opt in with certainty.
```

Comment:
What's the advice if there are changes?

Reply:
The reason do this validation is for users to be certain that no resolutions have changed, particularly for large projects — it won't at this phase reveal any inaccessible imports.

e.g. Things like a `"browser"` conditional import now being resolved correctly. This might cause a subtle, but investigable change:

```diff
- node_modules/some-pkg/src/index.js
+ node_modules/some-pkg/dist/index.cjs
```

For brevity, I haven't included advice beyond "see if resolution changes occur", given the above output (should be both 🤞🏻) 1/ rare, and 2/ instructive.